modules/calib3d/src/usac.hpp

```diff
@@ -125,35 +133,54 @@ class AffineMinimalSolver : public MinimalSolver {
     static Ptr<AffineMinimalSolver> create(const Mat &points_);
 };
 
+class TrifocalTensorMinimalSolver : public MinimalSolver {
+public:
+    static Ptr<TrifocalTensorMinimalSolver> create(const Mat &points_);
+    virtual void getFundamentalMatricesFromTensor (const cv::Mat &tensor, cv::Mat &F21, cv::Mat &F31) = 0;
+};
+
 //////////////////////////////////////// NON MINIMAL SOLVER ///////////////////////////////////////
 class NonMinimalSolver : public Algorithm {
 public:
+    virtual int estimate (const Mat &/*model*/, const std::vector<int> &sample, int sample_size, std::vector<Mat>
+            &models, const std::vector<double> &weights) const {
+        return estimate(sample, sample_size, models, weights);
+    }
     // Estimate models from non minimal sample. models.size() == number of found solutions
     virtual int estimate (const std::vector<int> &sample, int sample_size,
           std::vector<Mat> &models, const std::vector<double> &weights) const = 0;
     // return minimal sample size required for non-minimal estimation.
     virtual int getMinimumRequiredSampleSize() const = 0;
     // return maximum number of possible solutions.
     virtual int getMaxNumberOfSolutions () const = 0;
-    virtual Ptr<NonMinimalSolver> clone () const = 0;
+    virtual int estimate (const std::vector<bool> &/*mask*/, std::vector<Mat> &/*models*/,
```

Comment:
why not `virtual int estimate (..)= 0`?

Reply:
the function is quite sparse and not used in many classes. However, I made it virtual.